enginetest/queries/json_table_queries.go

```diff
@@ -470,6 +478,307 @@ var JSONTableScriptTests = []ScriptTest{
+				Expected: []sql.Row{
+					{nil},
+				},
+			},
+			{
+				Query: "SELECT * FROM JSON_TABLE('{\"c1\":\"abc\"}', '$' COLUMNS(c1 INT PATH '$.c1' DEFAULT '123' ON ERROR)) as jt;",
+				Expected: []sql.Row{
+					{123},
+				},
+			},
+			{
+				Query:          "SELECT * FROM JSON_TABLE('{\"c1\":\"abc\"}', '$' COLUMNS(c1 INT PATH '$.c1' DEFAULT 'def' ON ERROR)) as jt;",
+				ExpectedErrStr: "error: 'def' is not a valid value for 'int'",
+			},
+		},
+	},
+	{
+		Name:        "test DEFAULT ON EMPTY",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM JSON_TABLE('{}', '$' COLUMNS(c1 INT PATH '$.c1' DEFAULT '123' ON EMPTY)) as jt;",
+				Expected: []sql.Row{
+					{123},
+				},
+			},
+			{
+				Query: "SELECT * FROM JSON_TABLE('{\"notc1\": \"321321\"}', '$' COLUMNS(c1 INT PATH '$.c1' DEFAULT '123' ON EMPTY)) as jt;",
+				Expected: []sql.Row{
+					{123},
+				},
+			},
+			{
+				// MySQL only supports string type for DEFAULT
+				Query: "SELECT * FROM JSON_TABLE('{}', '$' COLUMNS(c1 INT PATH '$.c1' DEFAULT 123 ON EMPTY)) as jt;",
+				Expected: []sql.Row{
+					{123},
+				},
+			},
+		},
+	},
+	{
+		Name:        "test ERROR ON ERROR",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM JSON_TABLE('{}', '$' COLUMNS(c1 INT PATH '$.c1' ERROR ON ERROR)) as jt;",
+				Expected: []sql.Row{
+					{nil},
+				},
+			},
+			{
+				Query:          "SELECT * FROM JSON_TABLE('{\"c1\":\"abc\"}', '$' COLUMNS(c1 INT PATH '$.c1' ERROR ON ERROR)) as jt;",
+				ExpectedErrStr: "error: 'abc' is not a valid value for 'int'",
+			},
+		},
+	},
+	{
+		Name:        "test ERROR ON EMPTY",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:          "SELECT * FROM JSON_TABLE('{}', '$' COLUMNS(c1 INT PATH '$.c1' ERROR ON EMPTY)) as jt;",
+				ExpectedErrStr: "missing value for JSON_TABLE column 'c1'",
+			},
+			{
+				Query:          "SELECT * FROM JSON_TABLE('{\"notc1\": \"321321\"}', '$' COLUMNS(c1 INT PATH '$.c1' ERROR ON EMPTY)) as jt;",
+				ExpectedErrStr: "missing value for JSON_TABLE column 'c1'",
+			},
+		},
+	},
+	{
+		Name:        "test NESTED simple",
+		SetUpScript: []string{},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM JSON_TABLE('[{\"a\": 1, \"b\": [11,111]}, {\"a\": 2, \"b\": [22,222]}]', '$[*]' COLUMNS(a INT PATH '$.a', NESTED PATH '$.b[*]' COLUMNS (b1 INT PATH '$'))) AS jt;",
+				Expected: []sql.Row{
+					{1, 11},
+					{1, 111},
+					{2, 22},
+					{2, 222},
+				},
+			},
+			{
+				Query: "SELECT * FROM  JSON_TABLE('[{\"a\": 1, \"b\": [11,111]}, {\"a\": 2, \"b\": [22,222]}]', '$[*]' COLUMNS( a INT PATH '$.a', NESTED PATH '$.b[*]' COLUMNS (b1 INT PATH '$', b2 INT PATH '$'))) AS jt;",
+				Expected: []sql.Row{
+					{1, 11, 11},
+					{1, 111, 111},
+					{2, 22, 22},
+					{2, 222, 222},
+				},
+			},
+			// TODO: there's something wrong with how we handle these
```

Comment:
Can you add more details here? Looks like this query returns the same results in MySQL. Would be helpful for this comment to be actionable for a future reader to help fix whatever the issue is.